src/Controls/src/Core/NavigationPage.cs

```diff
@@ -315,7 +326,7 @@ async Task<Page> INavigationPageController.RemoveAsyncInner(Page page, bool anim
 			var removed = true;
 
 			EventHandler<NavigationRequestedEventArgs> requestPop = PopRequested;
-			if (requestPop != null)
+			if (requestPop != null && !requestedFromHandler)
```

Comment:
If the handler just wants you to remove a page don't fire requestPop

This eliminates all those super annoying _isPopping type checks in the renderer